### PR TITLE
Change libswiftnav to support L2CM tracking in Peregrine

### PR DESCRIPTION
**Pull request by rge-exafore**
This pull request combines #315 and #326. Both those requests are required to merge Peregrine changes to support L2C tracking.

### CODE REVIEW

**Comment by adel-mamin** [comment]
The changes look good to me.

**Comment by rge-exafore** [comment]
HITL tests: piksi_firmware_v0.21-209-g7c56849.hex

File: python/swiftnav/nav_msg.pyx [hunk 1]
```diff
@@ -42,7 +41,8 @@ cdef class NavMsg:
 
   def process_subframe(self, sid, d):
     cdef gps_l1ca_decoded_data_t tmp_d = d._thisptr
-    return process_subframe(&self._thisptr, sid, &tmp_d)
+    cdef gnss_signal_t tmp_sid = sid._thisptr
+    return process_subframe(&self._thisptr, tmp_sid, &tmp_d)
```
Comment (valeri-atamaniouk) [inline python/swiftnav/nav_msg.pyx]: tmp_d is a local. so no data is returned in d.

**Comment by gsmcmullin** [comment]
I don't see any cause for concern.  Merge if you're done with it.

**Pull request merged**